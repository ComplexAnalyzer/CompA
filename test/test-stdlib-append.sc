/*def appendi(arr1, arr2, arr, len1, len2) {
    tmp1 = len1;
    tmp2 = len2; 
    (0<=i<len1) arr[i] = arr1[i];
    (0<=i<len2) { j = i + len1; arr[j] = arr2[i]; }
    return arr;
}*/

def main() {
    a = [1,2,3];
    b = [4,5];
    c = [0, 0, 0, 0, 0];
    print(a[1]);
    (0<=i<3) print(a[i]);
    (0<=i<2) print(b[i]);
    (0<=i<5) print(c[i]);
    print("call");
    appendi(a, b, c, 3, 2);
    print("after");
    (0<=i<5) print(c[i]);
    print("here");
    d = a + b;
    (0<=i<5) print(c[i]); 
}
