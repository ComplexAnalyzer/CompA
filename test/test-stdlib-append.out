2
1
2
3
4
5
0
0
0
0
0
call
after
1
2
3
4
5
here
1
2
3
4
5
